engines/mxnet/jnarator/build.gradle

```diff
@@ -33,3 +33,5 @@ jar {
     duplicatesStrategy = DuplicatesStrategy.INCLUDE
     from configurations.runtimeClasspath.collect { it.isDirectory() ? it : zipTree(it) }
 }
+
+generateGrammarSource.finalizedBy verifyJava
```

Comment:
why is this a `finalizedBy` relationship and not a `dependsOn` from `verifyJava`?

Reply:
My reasoning here was that `verifyJava` was in a centralized gradle file, and adding all the required dependencies there would mean that over time that file could grow to know about a lot of other tasks.

As such I inverted the relationship so that the tasks that are distributed in the modules finalize the common tasks. In theory, this means that when adding a new module, you only need to worry about your module's build file, not any others. 

In theory 😄

Reply:
So, if you run `verifyJava`, does the `generateGrammarSource` also run? [Here](https://discuss.gradle.org/t/when-should-one-use-dependson-and-when-finalizedby/4066/2) is a nice explanation when to use which ordering, and [here](https://docs.gradle.org/current/userguide/controlling_task_execution.html) are the docs on task ordering and depednencies.

Reply:
Ah, I see what was meant originally.... it should indeed be a `dependsOn`. I will fix it.

I misunderstood the comment and thought you were referring to why I added the `finalizedBy` to this task instead of where verifyJava was declared 🙃

Reply:
Fixed - there was a hidden dependency of tasks that I 'patched' with `finalizedBy`, but it really should have been a chain of `dependsOn`. 

Thanks for the comments!